plugin/kubernetes/handler.go

```diff
@@ -44,14 +44,14 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 		records, extra, err = plugin.SRV(ctx, &k, zone, state, plugin.Options{})
 	case dns.TypeSOA:
 		records, err = plugin.SOA(ctx, &k, zone, state, plugin.Options{})
+	case dns.TypeAXFR, dns.TypeIXFR:
+		k.Transfer(ctx, state)
 	case dns.TypeNS:
 		if state.Name() == zone {
 			records, extra, err = plugin.NS(ctx, &k, zone, state, plugin.Options{})
 			break
 		}
-		fallthrough
-	case dns.TypeAXFR, dns.TypeIXFR:
-		k.Transfer(ctx, state)
+		_, err = parseRequest(state)
```

Comment:
why this?
the fallthrough was here to do nodata/nxdomain for ns queries other than the apex

Reply:
Yeah, I know _why_ it was there. But it wasn't working, (and there were no tests to validate that case) - it always returns NODATA, regardless of whether or not the subdomain exists.

Perhaps it makes better sense to restore the fallthrough, and then fix that issue in a separate PR.